loader/lightning.go

```diff
@@ -0,0 +1,307 @@
+			return err
+		}
+		l.Lock()
+		taskCtx, cancel := context.WithCancel(ctx)
+		l.cancel = cancel
+		l.Unlock()
+		err = l.core.RunOnce(taskCtx, cfg, nil)
+		if err == nil {
+			l.finish.Store(true)
+			offsetSQL := l.checkPoint.GenSQL(lightningCheckpointFile, 1)
+			err = l.toDBConns[0].executeSQL(tctx, []string{offsetSQL})
+			_ = l.checkPoint.UpdateOffset(lightningCheckpointFile, 1)
+		}
+	} else {
+		l.finish.Store(true)
+	}
+	if l.cfg.Mode == config.ModeFull {
+		if err = delLoadTask(l.cli, l.cfg, l.workerName); err != nil {
+			return err
+		}
+	}
+	if l.finish.Load() {
+		if l.cfg.CleanDumpFile {
+			cleanDumpFiles(l.cfg)
+		}
+	}
+	return err
+}
+
+// Process implements Unit.Process.
+func (l *LightningLoader) Process(ctx context.Context, pr chan pb.ProcessResult) {
+	l.logger.Info("lightning load start")
+	errs := make([]*pb.ProcessError, 0, 1)
+	binlog, gtid, err := getMydumpMetadata(l.cli, l.cfg, l.workerName)
+	if err != nil {
+		loaderExitWithErrorCounter.WithLabelValues(l.cfg.Name, l.cfg.SourceID).Inc()
+		pr <- pb.ProcessResult{
+			Errors: []*pb.ProcessError{unit.NewProcessError(err)},
+		}
+		return
+	}
+	if binlog != "" {
+		l.metaBinlog.Store(binlog)
+	}
+	if gtid != "" {
+		l.metaBinlogGTID.Store(gtid)
+	}
+
+	if err := l.restore(ctx); err != nil && !utils.IsContextCanceledError(err) {
+		errs = append(errs, unit.NewProcessError(err))
+	}
+	isCanceled := false
+	select {
+	case <-ctx.Done():
+		isCanceled = true
+	default:
+	}
+	l.logger.Info("lightning load end")
+	pr <- pb.ProcessResult{
+		IsCanceled: isCanceled,
+		Errors:     errs,
+	}
+}
+
+func (l *LightningLoader) isClosed() bool {
+	return l.closed.Load()
+}
+
+// IsFreshTask implements Unit.IsFreshTask.
+func (l *LightningLoader) IsFreshTask(ctx context.Context) (bool, error) {
+	count, err := l.checkPoint.Count(tcontext.NewContext(ctx, l.logger))
+	return count == 0, err
+}
+
+// Close does graceful shutdown.
+func (l *LightningLoader) Close() {
+	l.Pause()
+	l.closed.Store(true)
+}
+
+// Pause pauses the process, and it can be resumed later
+// should cancel context from external.
+func (l *LightningLoader) Pause() {
+	l.Lock()
+	defer l.Unlock()
+	if l.isClosed() {
+		l.logger.Warn("try to pause, but already closed")
+		return
+	}
+	if l.cancel != nil {
+		l.cancel()
+	}
+	l.core.Stop()
+}
+
+// Resume resumes the paused process.
+func (l *LightningLoader) Resume(ctx context.Context, pr chan pb.ProcessResult) {
+	if l.isClosed() {
+		l.logger.Warn("try to resume, but already closed")
+		return
+	}
+	l.core = lightning.New(l.lightningConfig)
+	// continue the processing
+	l.Process(ctx, pr)
+}
+
+// Update implements Unit.Update
+// now, only support to update config for routes, filters, column-mappings, block-allow-list
+// now no config diff implemented, so simply re-init use new config
+// no binlog filter for loader need to update.
+func (l *LightningLoader) Update(cfg *config.SubTaskConfig) error {
+	// update l.cfg
+	l.cfg.BAList = cfg.BAList
```

Comment:
How does these configs take effect. After table checkpoint has been inited, these config change won't take effect, does this behavior in line with expectation?